crates/sp-domains-fraud-proof/src/host_functions.rs

```diff
@@ -355,6 +355,13 @@ where
                 .map(|is_inherent| {
                     FraudProofVerificationInfoResponse::InherentExtrinsicCheck(is_inherent)
                 }),
+            FraudProofVerificationInfoRequest::ExtrinsicDecodableCheck(opaque_extrinsic) => {
+                let encoded_extrinsic = opaque_extrinsic.encode();
+                Some(FraudProofVerificationInfoResponse::ExtrinsicDecodableCheck(
+                    <DomainBlock as BlockT>::Extrinsic::decode(&mut encoded_extrinsic.as_slice())
```

Comment:
This decode would work until we have a single Domain type which is EVM. We use DomainBlock to access hash functions and other specific details rather than using the domain runtime. This assumption would work until we introduce a new domain type. What ideally should happen here is to use StateLess runtime with api that can take encoded extrinsic and runtime would try to decode this. This way, if we introduce new domain runtime types, each individual extrinsic is decoded per its own runtime rather than provided `DomainBlock` type

Reply:
>We use DomainBlock to access hash functions and other specific details rather than using the domain runtime. This assumption would work until we introduce a new domain type.

For the domain block body (i.e. extrinsic), it may vary depending on runtime, since different runtime may support different pallets and extrinsic. But for the domain block header, all domain runtime should be the same as `domain_runtime_primitives::opaque::Header` since the consensus runtime already uses `T::DomainHeader` and it can't support a different one.

Reply:
> But for the domain block header, all domain runtime should be the same as domain_runtime_primitives::opaque::Header since the consensus runtime already uses T::DomainHeader and it can't support a different one.

Yes and that is why it is okay to have T::DomainHeader for all the domains but not T::DomainBlock